pym/bob/cmds/build.py

```diff
@@ -458,15 +442,43 @@ def _runShell(self, step, scriptName):
         elif self.__verbose >= 2:
             cmdLine.append('-vv')
 
-        proc = subprocess.Popen(cmdLine, cwd=step.getWorkspacePath(), env=runEnv)
+        oup = pty.openpty()
+        erp = pty.openpty()
+
+        logFile = open(os.path.join(step.getWorkspacePath(), '..', 'log.txt'), 'a')
+        proc = subprocess.Popen(cmdLine, cwd=step.getWorkspacePath(), env=runEnv,
+                universal_newlines=True, stdout=oup[1], stderr=erp[1])
+
+        ansi_escape = re.compile(r'\x1B\[([0-9]{1,2}(;[0-9]{1,2})?)?[m|K]')
+
         try:
-            if proc.wait() != 0:
+            while 1:
+                timeout = .1
```

Comment:
Instead of a timeout a SIGCHLD handler should be used to get an immediate feedback and prevent polling.